csharp/ql/src/utils/model-generator/internal/CaptureModelsSpecific.qll

```diff
@@ -16,14 +17,25 @@ module TaintTracking = CS::TaintTracking;
 
 class Type = CS::Type;
 
+/**
+ * Holds if `api` is an override of `GetHashCode` or `Equals`.
+ */
+private predicate isIrrelevantObjectOveride(CS::Callable api) {
+  exists(System::SystemObjectClass c |
+    api = c.getGetHashCodeMethod().getAnOverrider*() or
+    api = c.getEqualsMethod().getAnOverrider*()
+  )
```

Comment:
In order to make this more robust towards future additions (like [`IEquatable<T>.Equals(T)`](https://docs.microsoft.com/en-us/dotnet/api/system.iequatable-1.equals?view=net-6.0)), I would rewrite it slightly as
```ql
exists(Callable exclude |
  api = exclude.getAnOverrider*().getUnboundDeclaration() or
  api = exclude.getAnUltimateImplementor().getUnboundDeclaration()
|
  exists(System::SystemObjectClass c |
    exlude = [c.getGetHashCodeMethod(), c.getEqualsMethod()]
  )
)
```

Reply:
Yes, it is a good idea to also exclude IEquatable<T>.Equals.